pkg/parser/parser.go

```diff
@@ -83,6 +85,7 @@ type dependencyGraphParser func(lockfilePath string, config *ParserConfig) (*mod
 
 // Maintain a map of lockfileAs to dependencyGraphParser
 var dependencyGraphParsers map[string]dependencyGraphParser = map[string]dependencyGraphParser{
+	".terraform.lock.hcl":             parseTerraformLockfile,
```

Comment:
This looks a bit odd. May be we should just use `customParserTerraform` here instead of `.terraform.lock.hcl`. This map also is used when I do something like:

```
vet scan -D /path/to/dir --type terraform
```

The `type` arg is used to lookup the "explicitly provided" parser to parse input files. However, when such explicit type is missing, then we need a way to find the parser based on file type. Lets figure that out as a future enhancement. May be you can file an issue for the future.

Reply:
so finally should we make a change to this ?

Reply:
@insaaniManav Set this as `customParserTerraform` for this PR.